timelock-impl/src/main/java/com/palantir/atlasdb/timelock/adjudicate/FeedbackProvider.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.adjudicate;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.palantir.common.streams.KeyedStream;
+import com.palantir.timelock.feedback.ConjureTimeLockClientFeedback;
+import com.palantir.timelock.feedback.EndpointStatistics;
+
+public final class FeedbackProvider {
+
+    private FeedbackProvider() {
+        // no op
+    }
+
+    public static HealthStatus getTimeLockHealthStatus(TimeLockClientFeedbackSink timeLockClientFeedbackSink) {
+        List<ConjureTimeLockClientFeedback> trackedFeedbackReports =
+                timeLockClientFeedbackSink.getTrackedFeedbackReports();
+
+        Map<String, Map<UUID, List<ConjureTimeLockClientFeedback>>> organizedFeedback =
```

Comment:
While this is fine, it's usually preferable to have structures for the intermediate layers, because it makes things easier to reason about. So I'd think

- the inner one could be a `Multimap<UUID, List<ConjureTimeLockClientFeedback>>`. But really this is a mapping of nodes to feedback reports. I might call this intermediate structure `ServiceFeedback`.
- the top level concept here as a `String` is a service name. I'd consider naming it maybe `organizeFeedbackReportsByService` and/or have a string wrapping class